src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -632,6 +633,125 @@ public void canPlaceAnItemInTransit()
     assertThat(tags, hasItem(TAG_VALUE));
   }
 
+  @Test
+  public void checkIfStatusDateExistsWhenItemStatusUpdated()
+    throws MalformedURLException, InterruptedException,
+    ExecutionException, TimeoutException {
+
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    UUID id = UUID.randomUUID();
+    JsonObject itemToCreate = smallAngryPlanet(id, holdingsRecordId);
+
+    createItem(itemToCreate);
+
+    final String inTransitServicePointId = UUID.randomUUID().toString();
+
+    JsonObject replacement = itemToCreate.copy();
+
+    replacement
+      .put("status", new JsonObject().put("name", "Checked out"));
+
+    CompletableFuture<Response> replaceCompleted = new CompletableFuture<>();
+
+    client.put(itemsStorageUrl(String.format("/%s", id)), replacement,
+      StorageTestSuite.TENANT_ID, ResponseHandler.empty(replaceCompleted));
+
+    Response putResponse = replaceCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(putResponse.getStatusCode(), is(HttpURLConnection.HTTP_NO_CONTENT));
+
+    Response getResponse = getById(id);
+
+    assertThat(getResponse.getStatusCode(), is(HttpURLConnection.HTTP_OK));
+
+    JsonObject item = getResponse.getJson();
+
+    assertThat(item.getString("id"), is(id.toString()));
+
+    assertThat(item.getJsonObject("status").getString("name"),
+      is("Checked out"));
+
+    assertThat(item.getJsonObject("status").getString("date"),
+      notNullValue());
```

Comment:
Can we check for more than not null? Since we are relying on PostgreSQL to create and store the date with the proper format, can we at least verify the date is ISO8601 in this test?

Reply:
@mreno-EBSCO done it. Could you please review my approach?